lib/flatirons/saas/services/stripe_service.rb

```diff
@@ -90,6 +90,12 @@ def update_subscription(subscription_id, price_id, proration_behavior = 'always_
       subscription
     end
 
+    def delete_subscription(subscription_id)
+      return unless subscription_id
+
+      Stripe::Subscription.delete(subscription_id)
```

Comment:
`Stripe::Subscription.delete(subscription_id, {}, stripe_opts)` **stripe_opts**

Add the `stripe_opts` because we are not using the global stripe_key.

Add the fallow options:

- **invoice_now** optional
Will generate a final invoice that invoices for any un-invoiced metered usage and new/pending proration invoice items.

- **prorate** optional
Will generate a proration invoice item that credits remaining unused time until the subscription period end.


```ruby
def delete_subscription(subscription_id, invoice_now: false, prorate: false)
      return unless subscription_id

      Stripe::Subscription.delete(subscription_id, {invoice_now: invoice_now, prorate: prorate}, stripe_opts)
end
```